src/game/server/player.h

```diff
@@ -622,6 +628,7 @@ class CBasePlayer : public CBaseCombatCharacter
 	// say/sayteam allowed?
 	virtual bool		CanHearAndReadChatFrom( CBasePlayer *pPlayer ) { return true; }
 	virtual bool		CanSpeak( void ) { return true; }
+	virtual PlayerSpeakResult        CheckSpeak( void );
```

Comment:
Unless we're planning on making this a lot more complicated very soon, I'd prefer that this just returns a bool.

Reply:
should just edit the canspeak function instead of a new one